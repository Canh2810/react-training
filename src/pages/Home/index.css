.home {
  background-color: var(--background-primary-color);
}

.home__content {
  margin: 80px auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 32px;
}
