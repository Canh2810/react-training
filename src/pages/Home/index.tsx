// Libs
import { memo, useCallback, useState } from 'react'

// Types
import { ButtonVariants } from '@types'

// Components
import {
  Advertisement,
  Banner,
  Button,
  LoadingIndicator,
  PostList,
  SearchResults,
  Typography,
} from '@components'

// Constants
import { LIMIT_POSTS } from '@constants'

// Stores
import { usePostsStore, useQueryStore } from '@stores'

// Style
import './index.css'
import { useFetchPosts, useDebounce, useFetchUser } from '@hooks'

const Home = () => {
  const [limitPosts, setLimitPosts] = useState<string>(LIMIT_POSTS)
  const { posts } = usePostsStore((state) => state)
  const { query } = useQueryStore((state) => state)
  const debouncedValue = useDebounce<string>(query)
  const {
    isLoading: isLoadingFetchPosts,
    isError: isErrorFetchPosts,
    error: errorFetchPosts,
  } = useFetchPosts(limitPosts, debouncedValue)
  const {
    isLoading: isLoadingFetchUser,
    isError: isErrorFetchUser,
    error: errorFetchUser,
  } = useFetchUser()

  const handleToggleViewALlPosts = useCallback(() => {
    setLimitPosts((prev) => (prev === LIMIT_POSTS ? '' : LIMIT_POSTS))
  }, [setLimitPosts])

  const buttonTitle = limitPosts === LIMIT_POSTS ? 'View all post' : 'Hide less'
  const isLoading = isLoadingFetchPosts || isLoadingFetchUser
  const isError = isErrorFetchPosts || isErrorFetchUser
  const errorMessage = errorFetchPosts?.message || errorFetchUser?.message

  return (
    <div className="home">
      {isError && <Typography>{errorMessage}</Typography>}
      {debouncedValue.length > 0 ? (
        <SearchResults posts={posts} />
      ) : (
        <>
          <Banner post={posts[0]} />
          <Advertisement />
          <section className="home__content">
            <PostList posts={posts} />
            <Button
              variant={ButtonVariants.Outlined}
              onClick={handleToggleViewALlPosts}
            >
              {buttonTitle}
            </Button>
          </section>
          <Advertisement />
        </>
      )}
      {isLoading && <LoadingIndicator />}
    </div>
  )
}

export default memo(Home)
