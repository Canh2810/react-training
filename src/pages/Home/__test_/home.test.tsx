import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import Home from '..'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { MemoryRouter } from 'react-router-dom'
import { POSTS_MOCK } from '@constants'
import { usePostsStore, useQueryStore } from '@stores'

const queryClient = new QueryClient()

const setup = () =>
  render(
    <QueryClientProvider client={queryClient}>
      <Home />
    </QueryClientProvider>,
    { wrapper: MemoryRouter },
  )

const mockUseFetchPosts = jest.fn()
const mockUseFetchUser = jest.fn()
const mockUseDebounce = jest.fn()

jest.mock('@hooks', () => ({
  ...jest.requireActual('@hooks'),
  useFetchPosts: () => mockUseFetchPosts,
  useFetchUser: () => mockUseFetchUser,
  useDebounce: () => mockUseDebounce,
}))

describe('Home page', () => {
  beforeEach(() => {
    mockUseFetchPosts.mockReturnValue({
      isLoading: false,
      isError: false,
      error: null,
    }),
      mockUseFetchUser.mockReturnValue({
        isLoading: false,
        isError: false,
        error: null,
      }),
      mockUseDebounce.mockReturnValue(''),
      usePostsStore.setState({
        posts: POSTS_MOCK,
      }),
      useQueryStore.setState({
        query: '',
      })
  })
  it('should render correctly', () => {
    const { container } = setup()
    expect(container).toMatchSnapshot()
  })

  it('should toggle view all posts', () => {
    setup()
    const button = screen.getByText('View all post')

    expect(button).toBeInTheDocument()

    fireEvent.click(button)

    expect(button).toHaveTextContent('Hide less')
  })

  it('should show error message if there is an error while fetching posts', () => {
    mockUseFetchPosts.mockReturnValue({
      isLoading: false,
      isError: true,
      error: { message: 'Error fetching posts' },
    })

    setup()
    waitFor(() => {
      expect(screen.getByText('Error fetching posts')).toBeInTheDocument()
    })
  })

  it('should show loading indicator while fetching data', () => {
    mockUseFetchPosts.mockReturnValue({
      isLoading: true,
      isError: false,
      error: null,
    })
    mockUseFetchUser.mockReturnValue({
      isLoading: true,
      isError: false,
      error: null,
    })

    setup()
    waitFor(() => {
      expect(screen.getByTestId('loading-indicator')).toBeInTheDocument()
    })
  })

  it('should render search results when there is a query', () => {
    mockUseDebounce.mockReturnValue('abc')

    waitFor(() => {
      expect(screen.getByTestId('No Result Found')).toBeInTheDocument()
    })
  })
})
