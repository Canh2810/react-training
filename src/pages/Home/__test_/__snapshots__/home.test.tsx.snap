// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly 1`] = `
<div>
  <div
    class="home"
  >
    <section
      class="banner"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="banner__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="banner__card"
      >
        <span
          class="badge"
          style="background-color: rgb(75, 107, 251); color: rgb(255, 255, 255);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 36px; font-weight: 600; line-height: 40px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="banner__card-footer"
        >
          <img
            alt="Tracey Wilson"
            class="banner__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: plus-jakarta-sans; font-size: 16px; font-weight: 500; line-height: 24px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: plus-jakarta-sans; font-size: 16px; font-weight: 500; line-height: 24px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </section>
    <div
      class="advertisement"
    >
      <p
        class=""
        style="font-family: inter; font-size: 14px; font-weight: 400; line-height: 20px; color: rgb(59, 60, 74);"
      >
        Advertisement
      </p>
      <p
        class=""
        style="font-family: work-sans; font-size: 20px; font-weight: 600; line-height: 28px; color: rgb(105, 106, 117);"
      >
        You can place ads
      </p>
      <p
        class=""
        style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
      >
        750x100
      </p>
    </div>
    <section
      class="home__content"
    >
      <div
        class="post-list"
      >
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Beta Testing: The final decisive step before launching the software"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Beta Testing: The final decisive step before launching the software
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Canh Vo"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEJuOqDYeLdcJOq8OctOVOCWp-RTFkdmWK4dcI5F9YLg&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Canh Vo
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Make a placeholder content in SwiftUI with Redacted & Unredacted"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Make a placeholder content in SwiftUI with Redacted & Unredacted
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSHeu1JU1avFzQtN-az4HZqEpR1VvEBN-SEXfEuOtt8Xg&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Instructions for Security Testing using OWASP ZAP tool for penetration testers - Penetration testing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Instructions for Security Testing using OWASP ZAP tool for penetration testers - Penetration testing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Excel at Cashiering with the McDonald POS Training Simulator APK!
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1418489098061-ce87b5dc3aee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQIgYqjNVQSYlcOONsKMfyvLJWY-MGgtqwfqMx7-2kjzg&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Excel at Cashiering with the McDonald POS Training Simulator APK!
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRlHJmFah1lW88FU0D-fD9m-0HXQGHd5QfD8GiL4guU-g&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1418489098061-ce87b5dc3aee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://www.w3schools.com/w3images/avatar2.png"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="The Impact of Technology on the Workplace: How Technology is Changing"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              The Impact of Technology on the Workplace: How Technology is Changing
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSHeu1JU1avFzQtN-az4HZqEpR1VvEBN-SEXfEuOtt8Xg&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
        <div
          class="post-card"
        >
          <img
            alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
            class="post-card__img"
            src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
          />
          <div
            class="post-card__content"
          >
            <span
              class="badge"
              style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
            >
              Technology
            </span>
            <p
              class=""
              style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
            >
              Excel at Cashiering with the McDonald POS Training Simulator APK!
            </p>
            <div
              class="post-card__footer"
            >
              <img
                alt="Tracey Wilson"
                class="post-card__author-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEJuOqDYeLdcJOq8OctOVOCWp-RTFkdmWK4dcI5F9YLg&s"
              />
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                Tracey Wilson
              </p>
              <p
                class=""
                style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
              >
                August 20, 2022
              </p>
            </div>
          </div>
        </div>
      </div>
      <button
        class="btn "
        style="border: 1px solid; color: rgb(105, 106, 117); border-color: rgba(105, 106, 117, 0.30); background-color: rgb(255, 255, 255);"
      >
        View all post
      </button>
    </section>
    <div
      class="advertisement"
    >
      <p
        class=""
        style="font-family: inter; font-size: 14px; font-weight: 400; line-height: 20px; color: rgb(59, 60, 74);"
      >
        Advertisement
      </p>
      <p
        class=""
        style="font-family: work-sans; font-size: 20px; font-weight: 600; line-height: 28px; color: rgb(105, 106, 117);"
      >
        You can place ads
      </p>
      <p
        class=""
        style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
      >
        750x100
      </p>
    </div>
  </div>
</div>
`;
